src/coreclr/vm/fcall.h

```diff
@@ -799,6 +799,22 @@ LPVOID __FCThrowArgument(LPVOID me, enum RuntimeExceptionKind reKind, LPCWSTR ar
 #define HELPER_METHOD_FRAME_GET_RETURN_ADDRESS()                                        \
     ( static_cast<UINT_PTR>( (__helperframe.InsureInit(NULL)), (__helperframe.MachineState()->GetRetAddr()) ) )
 
+#define EXCEPTION_METHOD_FRAME_BEGIN(funCallDepth) \
+        do \
+        { \
+            FrameWithCookie<SoftwareExceptionFrame> __exceptionFrame; \
+            *(&__exceptionFrame)->GetGSCookiePtr() = GetProcessGSCookie(); \
```

Comment:
Is this needed? `FrameWithCookie` constructor should initialize the GSCookie

Reply:
Yes, it is needed. There is the same thing with the FaultingExceptionFrame, it was doing the same thing forever. The GetGSCookiePtr() gets the ptr by subtracting 8 from the address of the frame. The FrameWithCookie stores the cookie as the first field and the actual frame as the second one. But the FaultingExceptionFrame and SoftwareExceptionFrame need to be aligned to 16 bytes, so the cookie that the FrameWithCookie sets is at a different location.

Reply:
Ah ok.